test/Test/Pos/Ssc/GodTossing/ComputeSharesSpec.hs

```diff
@@ -87,81 +89,36 @@ testMpcThdPortition = unsafeCoinPortionFromDouble testMpcThd
 isLimitedBy :: Word16 -> SharesDistribution -> Bool
 isLimitedBy mx sd = sum (toList sd) <= mx
 
--- This type describes the
--- (max inaccuracy, number of bad cases of the first type, number of bad cases of the second type)
--- More details see in the @isDistrFair@ description.
-type Unfairness = (Rational, Int, Int)
-
--- We will call distribution fair, if the max inaccuracy is less than 0.05
--- and number of bad cases is less than 5% of all 2^n cases.
--- Max inaccuracy is more or less heuristic value
--- which means which inaccuracy we can get in the bad case.
--- This inaccuracy can lead to two bad situation:
--- 1. when nodes mustn't reveal commitment, but they can
--- 2. when nodes must reveal commitment, but they can't.
--- We can get these situations when sum of stakes of nodes which sent shares is close to 0.5.
--- Max inaccuracy is computed as difference between generated stake and 0.5.
--- There is way to estimate inaccuracy as difference between real distribution and
--- generated. On the one hand it makes sense but on the other hand this difference
--- is not so important.
+-- Distribution is fair if a max inaccuracy is less than @sharesDistrInaccuracy@.
 isDistrFair :: RichmenStakes -> SharesDistribution -> Bool
-isDistrFair rs sd
-    | length rs > 20 = error "Too many richmen"
-    | otherwise = do
-        let n = length rs
-        let totalCases = toRational (2::Int)^n :: Rational
-        let fivePerc = 0.05 :: Rational
-        let totalCoins = sum $ map unsafeGetCoin $ toList rs
-        let totalDistr = sum (toList sd)
-        let stakeholders = HM.keys rs
-        let distrs = map (findStk totalCoins totalDistr) stakeholders
-        let !(er, firstCase, secondCase) = fairBrute distrs 0 0
-        er < sharesDistrInaccuracy &&
-            toRational firstCase < totalCases * fivePerc &&
-            toRational secondCase < totalCases * fivePerc
+isDistrFair (HM.map unsafeGetCoin -> rs) sd = do
+    let !totalDistr = sum sd
+    let !totalCoins = sum rs
+    let stakeholders = HM.keys rs
+    let coinsNDistr = map (first fromIntegral . findStk) stakeholders
+    isDistrInaccuracyAcceptable (fromIntegral totalCoins, totalDistr) coinsNDistr
```

Comment:
The problem now is that we rely on `isDistrInaccuracyAcceptable` correctness, but we don't know whether it's correct.
It would be good to add some tests for it. Probably very simple, just hardcode some numbers (close to corner cases).